flink-tests/src/test/java/org/apache/flink/test/checkpointing/IgnoreInFlightDataITCase.java

```diff
@@ -179,11 +205,30 @@ public void initializeState(FunctionInitializationContext context) throws Except
 
     private static class SlowMap extends RichMapFunction<Integer, Integer> {
 
+        private static OneShotLatch checkpointReachSinkLatch;
+        private static AtomicInteger idGenerator;
+        public int subtaskMapId;
+
+        public SlowMap() {
+            checkpointReachSinkLatch = new OneShotLatch();
+            idGenerator = new AtomicInteger();
```

Comment:
Constructor isn't called during the deserialization so it is called only once when we called constructor explicitly so there is no such problem.

Reply:
I see. To me, it would be more readable to initialize those fields not in constructor but in a static method called from the test body.

Or probably `SharedObjects` from #16064 can be used here.

Reply:
I took a deeper look and realized that we only need subtaskIndex:
```
        @Override
        public Integer map(Integer value) throws Exception {
            // Allow working only one last subtask until the checkpoint barrier reaches the sink.
            if (getRuntimeContext().getIndexOfThisSubtask() > 0) {
```
and don't need neither `idGenerator` nor `subtaskMapId`.